src/field.cpp

```diff
@@ -947,6 +948,7 @@ bool map::process_fields_in_submap( submap *const current_submap,
                                     one_in( 200 - cur->getFieldDensity() * 50 ) ) {
                                     ter_set( p, t_dirt );
                                     furn_set( p, f_ash );
+                                    add_item_or_charges( p, ash );
```

Comment:
Did you test it? I suspect it will not work due to SEALED flag.

While this type of code is used for rubble, it is not really the proper way to do it: the item always exists inside the ash/rubble instead of "being the ash/rubble". This will lead to problems if someone makes, for example, rain wash away the ash or something forcibly remove the ash some other way.

A more proper way would be to define what does it get cleared to somewhere (in the json, possibly just bash or deconstruct data) and then have that be placed under player's feet in `activity_handlers::clear_rubble_finish`.

Reply:
Yes, I tested it, and it worked fine. I chose this way because of its simplicity and minimal code changes.
I had an idea in my mind about removing `rubble` iuse action from `f_ash` and adding its own examine action - `e`xamining f_ash would just place `ash` at the f_ash tile and then remove f_ash.
Would my current working implementation suffice?